spec/controllers/registrations_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'Sign Up user via POST', type: :request do
+  it "should create User when given valid email, gender, password and password confirmation" do
+    expect {
+      post '/api/auth/users', params: { user: { email: 'test@mail.com', password: 'testing', password_confirmation: 'testing', gender: 'male' } }
+    }.to change { User.count }.by(1)
+
+    expect(response).to have_http_status(:ok)
+  end
+
+  it "should fail when given less-than-8-characters password" do
+    post '/api/auth/users', params: { user: { email: 'test@mail.com', password: 'testing', password_confirmation: 'test', gender: 'male' } }
```

Comment:
See https://relishapp.com/rspec/rspec-core/docs/subject/explicit-subject

Reply:
Change scopes and improve readability/performance. On it.